src/gsy_e/gsy_e_core/util.py

```diff
@@ -272,6 +269,10 @@ def convert_nested_settings(class_object, class_name, settings_dict):
         for key, value in dict(class_object.__dict__).items():
             if key.startswith("__"):
                 continue
+
+            if class_name not in settings_dict:
+                settings_dict[class_name] = {}
```

Comment:
These lines fix a bug with the way we were populating the `settings_dict`. The previous version of this method would fail if the configuration class (e.g. `MASettings`) did not have at least one parameter with values. So in a case like this:
```
class IAASettings(MASettings):
    pass
```
the method was raising exceptions. These two lines fix the issue.